staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -220,10 +220,15 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 
 	var handler http.HandlerFunc
+	subresources, err := apiextensions.GetCRDSubresourcesForVersion(&crd.Spec, requestInfo.APIVersion)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
use `response.WriteError(http.StatusInternalServerError, err)` instead to write a proper `metav1.Status`.